corehq/motech/openmrs/tasks.py

```diff
@@ -80,12 +80,29 @@ def get_openmrs_patients(requests, importer, location=None):
 
 
 def get_case_properties(patient, importer):
+    """
+    Returns case name and dictionary of case properties to update
+
+    Raises ConfigurationError if a value cannot be deserialized using
+    the data types given in a column mapping.
+    """
     name_columns = importer.name_columns.split(' ')
     case_name = ' '.join([patient[column] for column in name_columns])
-    fields_to_update = {
-        m.property: m.deserialize(patient[m.column], importer.get_timezone())
-        for m in importer.column_map
-    }
+    fields_to_update = {}
+    for mapping in importer.column_map:
+        try:
+            fields_to_update[mapping.property] = mapping.deserialize(
+                patient[mapping.column], importer.get_timezone()
+            )
+        except (TypeError, ValueError) as err:
+            raise ConfigurationError(
```

Comment:
the intent of putting the exception raise inside a loop it to determine at which step it failed I assume?

Reply:
Yes. We want to know exactly which value caused the problem. The problem is very likely that we thought the value was going to be one data type, but actually it's another.

... I see where your question might be going. ... Maybe we should build a list of all errors, and raise the exception after the loop. That way, if there is more than one column that isn't configured correctly, we can deal with all of them instead of just the first one.

... Good call.

Reply:
yeah. that would probably reduce the back and forth time of problem resolution later